pkg/settlement/pseudosettle/pseudosettle.go

```diff
@@ -72,6 +92,20 @@ func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (e
 
 	s.metrics.TotalReceivedPseudoSettlements.Add(float64(req.Amount))
 	s.logger.Tracef("received payment message from peer %v of %d", p.Address, req.Amount)
+
+	totalReceived, err := s.TotalReceived(p.Address)
+	if err != nil {
+		if err != ErrPeerNoSettlements {
+			return err
+		}
+		totalReceived = 0
```

Comment:
Good protection not to relay that with error a 0 value is returned. In general it is expected, but the safety here is nice to have, by my opinion.